### PR TITLE
Add Travis support.

### PR DESCRIPTION
**Pull request by vext01**
Travis test. Please ignore for now.

### CODE REVIEW

**Comment by vext01** [comment]
I don't know. I just copied and adapted the config from diffract.

We have more problems than that though. It looks like travis is too old for what we are doing with perf.

**Comment by vext01** [comment]
This isn't going to work I'm afraid. The newest environment travis can offer is Ubuntu Trusty, which is pre PT support.

Here's the manual page for `perf_event_open` in Trusty:
http://manpages.ubuntu.com/manpages/trusty/en/man2/perf_event_open.2.html

Note the lack of aux buffer.

**Comment by ltratt** [comment]
Does that mean we can't even get it to build?

**Comment by vext01** [comment]
Even `cargo check` wants to build the C part, so yes, sorry.

**Comment by vext01** [comment]
One hack would be to stub the C part if we are on travis.

**Comment by ltratt** [comment]
Is stubbing the C out difficult? If so, maybe we just have to skip Travis for this repo. Maybe we should set bors up on a build machine of our own...

**Comment by vext01** [comment]
We would basicallly put the whole contents of the C file inside an `#if !defined(TRAVIS)` and then try to convince cargo to pass `-D travis` to the C compiler if the `TRAVIS` environment is set.

Easy without Cargo, not sure about with.

I suggest I spend half an hour or so trying to figure this out before moving on. Plan?

**Comment by ltratt** [comment]
No argument from me.

**Comment by vext01** [comment]
Who likes to rock the party? :tada: 

I tried a few different things, this is the least foul of them. You OK with it?

Looking at the Travis logs, looks like travis doesn't run tests automatically.

**Comment by vext01** [comment]
(lol, half an hour. More like 2 hours).

File: build.rs [hunk 1]
```diff
@@ -36,10 +36,17 @@
...
-    gcc::Build::new()
-               .file("src/traceme.c")
-               .include("src")
-               .compile("traceme");
+    let mut c_build = gcc::Build::new();
+    c_build.file("src/traceme.c").include("src");
+
+    // Travis' Linux setup is too old for what we do with linux-perf.
+    // Setting -DTRAVIS causes the C API to be stubbed.
+    if env::var("TRAVIS").is_ok() {
```
Comment (ltratt) [inline build.rs]: Should (can?) we check *which* version of Travis? If/when they upgrade, maybe we should have an `assert` (or similar) to alert us of this, and make us investigate building things fully?

Comment (vext01) [reply build.rs]: We can check the output of `lsb_release -r` if we are on travis and panic if the version changed.

Comment (ltratt) [reply build.rs]: Sounds like a plan!

**Comment by vext01** [comment]
This seems to do the trick.

Passing on Travis. Here it is failing on bencher8 with `TRAVIS` in the env:
```
   Compiling traceme v0.1.0 (file:///home/vext01/research/traceme)
error: failed to run custom build command for `traceme v0.1.0 (file:///home/vext01/research/traceme)`
process didn't exit successfully: `/home/vext01/research/traceme/target/debug/build/traceme-0ac45483236dea7b/build-script-build` (exit code: 101)
--- stderr
thread 'main' panicked at 'assertion failed: lsb_rel["Distributor ID"] == "Ubuntu"', build.rs:69:8
```

Swapping the order of the assertions to test the other one:
```
   Compiling traceme v0.1.0 (file:///home/vext01/research/traceme)
error: failed to run custom build command for `traceme v0.1.0 (file:///home/vext01/research/traceme)`
process didn't exit successfully: `/home/vext01/research/traceme/target/debug/build/traceme-0ac45483236dea7b/build-script-build` (exit code: 101)
--- stderr
thread 'main' panicked at 'assertion failed: lsb_rel["Release"] == "14.04"', build.rs:69:8
```

File: build.rs [hunk 2]
```diff
@@ -37,14 +37,38 @@
...
 use std::env;
+use std::process::Command;
+use std::collections::HashMap;
+
+fn run_lsb_release(args: Vec<&str>) -> HashMap<String, String> {
+    let output = Command::new("lsb_release")
+                         .args(args)
+                         .output()
+                         .expect("failed to run lsb_release");
+    assert!(output.status.success(), "lsb_release failed");
```
Comment (ltratt) [inline build.rs]: Having one thing as an `expect` and one as an `assert` feels a bit odd? Maybe make both `expect`s and/-r unify the message? Or, to be honest, you could just `unwrap` them (I don't think putting `expect` everywhere is a good idea: it clogs up the source code).

Comment (vext01) [reply build.rs]: If you don't think it's favourable to include why we crashed, then we can just unwrap. Yeah. I got into this habit as backtraces rarely work(ed?) on BSD.

Comment (ltratt) [reply build.rs]: I think they might start working once the Rust port properly moves to LLVM and libunwind is fixed. Anyway, up to you.

**Comment by vext01** [comment]
Is this what you meant?

You can't unwrap() a bool, so it has to either be an assert or some other conditional check.

**Comment by ltratt** [comment]
I think so. If you're OK with it, please squash.

**Comment by vext01** [comment]
splat.

**Pull request merged**